Bio/Seq.py

```diff
@@ -395,7 +398,11 @@ def count_overlap(self, sub, start=0, end=sys.maxsize):
         HOWEVER, do not use this method for such cases because the
         count() method is much for efficient.
         """
-        sub_str = str(sub)
+        if isinstance(sub, (Seq, MutableSeq)):
+            sub_str = str(sub)
```

Comment:
or use `sub = str(sub)`, then we can drop the `else` statement.

Reply:
Yes, if we changed the while loop to use ``sub`` that would work. I thought this was clearer retaining the cast variables ``self_str`` and ``sub_str``.

Reply:
Up to you then. I am fine either way.

Reply:
I'll change it. They are both clear, but your proposal is shorter.